src/tools/tidy/src/ext_tool_checks/rustdoc_js.rs

```diff
@@ -0,0 +1,142 @@
+//! Tidy check to ensure that rustdoc templates didn't forget a `{# #}` to strip extra whitespace
+//! characters.
+
+use std::ffi::OsStr;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use ignore::DirEntry;
+
+use crate::walk::walk_no_read;
+
+fn rustdoc_js_files(librustdoc_path: &Path) -> Vec<PathBuf> {
+    let mut files = Vec::new();
+    walk_no_read(
+        &[&librustdoc_path.join("html/static/js")],
+        |path, is_dir| is_dir || !path.extension().is_some_and(|ext| ext == OsStr::new("js")),
+        &mut |path: &DirEntry| {
+            files.push(path.path().into());
+        },
+    );
+    return files;
+}
+
+fn run_eslint(args: &[PathBuf], config_folder: PathBuf) -> Result<(), super::Error> {
+    let mut child = Command::new("npx")
+        .arg("eslint")
+        .arg("-c")
+        .arg(config_folder.join(".eslintrc.js"))
+        .args(args)
+        .spawn()?;
+    match child.wait() {
+        Ok(exit_status) => {
+            if exit_status.success() {
+                return Ok(());
+            }
+            Err(super::Error::FailedCheck("eslint command failed"))
+        }
+        Err(error) => Err(super::Error::Generic(format!("eslint command failed: {error:?}"))),
+    }
+}
+
+fn get_eslint_version_inner(global: bool) -> Option<String> {
+    let mut command = Command::new("npm");
+    command.arg("list").arg("--parseable").arg("--long").arg("--depth=0");
+    if global {
+        command.arg("--global");
+    }
+    let output = command.output().ok()?;
+    let lines = String::from_utf8_lossy(&output.stdout);
+    lines.lines().find_map(|l| l.split(':').nth(1)?.strip_prefix("eslint@")).map(|v| v.to_owned())
+}
+
+fn get_eslint_version() -> Option<String> {
+    get_eslint_version_inner(false).or_else(|| get_eslint_version_inner(true))
+}
+
+pub(super) fn lint(
+    librustdoc_path: &Path,
+    tools_path: &Path,
+    src_path: &Path,
+) -> Result<(), super::Error> {
+    let eslint_version_path =
+        src_path.join("ci/docker/host-x86_64/mingw-check-tidy/eslint.version");
+    let eslint_version = match std::fs::read_to_string(&eslint_version_path) {
+        Ok(version) => version.trim().to_string(),
+        Err(error) => {
+            eprintln!("failed to read `{}`: {error:?}", eslint_version_path.display());
+            return Err(error.into());
+        }
+    };
+    // Having the correct `eslint` version installed via `npm` isn't strictly necessary, since we're invoking it via `npx`,
+    // but this check allows the vast majority that is not working on the rustdoc frontend to avoid the penalty of running
+    // `eslint` in tidy. See also: https://github.com/rust-lang/rust/pull/142851
+    match get_eslint_version() {
```

Comment:
I *do* think we can get rid of this check. I wanted to remove it in #142851 (in the first iteration, before I moved everything to be gated under the `--extra-checks` flag), but it would've made skipping `eslint` harder.
If we're going in the direction of only running `eslint` when needed, not just for everyone, then there's absolutely no reason to ask the user to explicitly install it, and not just use `npx` as it was intended to be used.
(just take care to run `npx eslint@<version we pinned>`)

Reply:
Yeah, but then most likely `npm` isn't either (unless someone is running a [really cursed environment](https://xkcd.com/1987/) [with `s/python/node/`] 😁)